amundsen_application/static/js/components/DashboardPage/QueryListItem/index.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import { OverlayTrigger, Popover } from 'react-bootstrap';
+
+import "./styles.scss";
+
+export interface QueryListItemProps {
+  text: string;
+  url: string;
+  name: string;
+}
+
+type GoToDashboardLinkProps = {
+  url: string;
+};
+
+const QUERY_LABEL = "Query";
+const MODE_LINK_TOOLTIP_TEXT = "View in Mode";
+const LOADING_QUERY_MESSAGE = "Loading Query Component, please wait...";
+
+const LazyComponent = React.lazy(() => import("./CodeBlock"));
+
+const GoToDashboardLink = ({ url }: GoToDashboardLinkProps) => {
+  const popoverHoverFocus = (<Popover id="popover-trigger-hover-focus">{MODE_LINK_TOOLTIP_TEXT}</Popover>);
+
+  return (
+    <OverlayTrigger
+      trigger={["hover", "focus"]}
+      placement="bottom"
+      overlay={popoverHoverFocus}
+    >
+      <a
+        className="query-list-query-link"
+        href={url}
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        <svg className="icon" fill="none" viewBox="0 0 24 24">
+          <path
+            fillRule="evenodd"
+            clipRule="evenodd"
+            d="M9 3v2.75H5.75v12.5h12.5V15H21v6H3V3h6zm12 0v9.75h-2.75V7.892L10.544 15.6 8.4 13.456l7.707-7.707-4.607.001V3H21z"
+          />
+        </svg>
+      </a>
+    </OverlayTrigger>
+  );
+};
+
+const QueryListItem = ({ name, text, url }: QueryListItemProps) => {
+  const [isExpanded, setExpanded] = React.useState(false);
+  const toggleExpand = () => {
+    setExpanded(!isExpanded);
+  };
+  const key = `key:${name}`;
+
+  return (
+    <li
+      className="list-group-item query-list-item clickable"
+      role="tab"
+      id={key}
+    >
+      <a
+        className="query-list-header"
+        aria-expanded={isExpanded}
+        aria-controls={key}
+        role="button"
+        href="#"
+        onClick={toggleExpand}
```

Comment:
I wonder if we should add this on the entire list item itself, but prevent the event from bubbling up from interactions withe the `GoToDashboardLink` or `CodeBlock`. When expanded, I would expect that any interaction with white space in the list item would close the item, but right now the user is required to maneuver back up to the top to close.

Reply:
Oh, interesting. I wasn't thinking that users would expect to close this item by clicking on any white space.

I know what you suggest is possible using pointer-events, but every time I have done it, it ended up being hacky and error prone. 

Happy to discuss it further with the team!

Reply:
Design would be the one to sync with now that I remember some previous context. From a technical perspective, I used to in the same boat with you when we added interactive elements inside of the column list items on `TableDetail`. I raised the technical caveats behind this, but design settled on expand/collapse target being the entire list item whitespace, plus this was something users preferred. So based on existing precedents I think we have to go this route for consistency, or have design be the ones to sign off on this choice.

Reply:
Just confirmed with Knowl, the current behavior is good!

Reply:
No problem. Knowl's approval is fine with me but I would say then that our designers may have to align on specific concepts amongst themselves given we've now received two different directions here for the same kind of interaction. 

I will leave a task for myself to raise this again when we have gotten past our higher priority design items, from the perspective of providing a consistent experience one way or the other.